clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -921,10 +938,11 @@ FunctionVariableConstraint::FunctionVariableConstraint(const Type *Ty,
           PName = PVD->getName();
         }
       }
-      bool IsGeneric =
-          ParmVD != nullptr && getTypeVariableType(ParmVD) != nullptr;
-      ParamVars.push_back(
-          new PVConstraint(QT, ParmVD, PName, I, Ctx, &N, IsGeneric));
+      auto *ParamVar = new PVConstraint(QT, ParmVD, PName, I, Ctx, &N);
+      int GenericIdx = ParamVar->getGenericIndex();
+      if (GenericIdx >= 0)
+        TypeParams = std::max(TypeParams, GenericIdx + 1);
```

Comment:
I don't understand this update to TypeParams

Reply:
`TypeParams` represents the number of variables in `Itype_for_any(...)` for a function. Unfortunately, I was unable to read this information directly from the decl. But, each parameter and return var has an index attached, so the max index (+1) is the number of variables. Eventually the rewriter will need to know the location of that phrase in the code, so if you're aware of it let me know. I could make this code slightly more efficient or save the notes for later.

Reply:
Makes sense. Changes all look took to me then.